analyzer/src/main/kotlin/managers/Bundler.kt

```diff
@@ -258,23 +292,46 @@ data class GemSpec(
 ) {
     companion object Factory {
         fun createFromYaml(spec: String): GemSpec {
-            val gemSpecTree = yamlMapper.readTree(spec)
+            val yaml = yamlMapper.readTree(spec)!!
 
-            val runtimeDependencies = gemSpecTree["dependencies"]?.asIterable()?.mapNotNull {
+            val runtimeDependencies = yaml["dependencies"]?.asIterable()?.mapNotNull {
                 if (it["type"]?.asText() == ":runtime")
                     it["name"]?.asText()
                 else
                     null
             }?.toSet()
 
             return GemSpec(
-                    gemSpecTree["name"].asText(),
-                    gemSpecTree["version"]["version"].asText(),
-                    gemSpecTree["homepage"].asText(),
-                    gemSpecTree["licenses"].asIterable().map { it.asText() }.toSortedSet(),
-                    gemSpecTree["description"].asText(),
+                    yaml["name"].asText(),
+                    yaml["version"]["version"].asText(),
+                    yaml["homepage"].asTextOrEmpty(),
+                    yaml["licenses"]?.asIterable()?.map { it.asText() }?.toSortedSet() ?: sortedSetOf(),
+                    yaml["description"].asTextOrEmpty(),
+                    runtimeDependencies ?: emptySet(),
+                    parseVcs(yaml["homepage"].asText())
+            )
+        }
+
+        fun createFromJson(spec: String): GemSpec {
+            val json = jsonMapper.readTree(spec)!!
+
+            val vcsUrl = json["source_code_uri"]?.asText()
+            val vcs: VcsInfo = if (vcsUrl.isNullOrBlank() || vcsUrl == "null") {  // FIXME
```

Comment:
FIXME?

Reply:
That actually gets fixed in the next commit.